File: src/main/java/com/miilhozinho/chunkgenerator/data/GenState.java
package com.miilhozinho.chunkgenerator.data;

import com.hypixel.hytale.codec.Codec;
import com.hypixel.hytale.codec.KeyedCodec;
import com.hypixel.hytale.codec.builder.BuilderCodec;

public class GenState {

    public static final BuilderCodec<GenState> CODEC = BuilderCodec.builder(GenState.class, GenState::new)
            .append(new KeyedCodec<String>("WorldName", Codec.STRING),
                    (state, value, extraInfo) -> state.WorldName = value,
                    (state, extraInfo) -> state.WorldName).add()
            .append(new KeyedCodec<Integer>("CenterX", Codec.INTEGER),
                    (state, value, extraInfo) -> state.centerX = value,
                    (state, extraInfo) -> state.centerX).add()
            .append(new KeyedCodec<Integer>("CenterZ", Codec.INTEGER),
                    (state, value, extraInfo) -> state.centerZ = value,
                    (state, extraInfo) -> state.centerZ).add()
            .append(new KeyedCodec<Integer>("TargetRadius", Codec.INTEGER),
                    (state, value, extraInfo) -> state.targetRadius = value,
                    (state, extraInfo) -> state.targetRadius).add()
            .append(new KeyedCodec<Integer>("CurrentIndex", Codec.INTEGER),
                    (state, value, extraInfo) -> state.currentIndex = value,
                    (state, extraInfo) -> state.currentIndex).add()
            .append(new KeyedCodec<Boolean>("IsPaused", Codec.BOOLEAN),
                    (state, value, extraInfo) -> state.isPaused = value,
                    (state, extraInfo) -> state.isPaused).add()
            .build();

    private String WorldName = "";
    private int centerX = 0;
    private int centerZ = 0;
    private int targetRadius = 0;
    private int currentIndex = 0;
    private boolean isPaused = true;

    public GenState() {
    }

    public GenState(String worldName, int centerX, int centerZ, int targetRadius) {
        this.WorldName = worldName;
        this.centerX = centerX;
        this.centerZ = centerZ;
        this.targetRadius = targetRadius;
        this.currentIndex = 0;
        this.isPaused = false;
    }

    public String getWorldName() {
        return WorldName;
    }

    public int getCenterX() {
        return centerX;
    }

    public int getCenterZ() {
        return centerZ;
    }

    public int getTargetRadius() {
        return targetRadius;
    }

    public int getCurrentIndex() {
        return currentIndex;
    }

    public void setCurrentIndex(int currentIndex) {
        this.currentIndex = currentIndex;
    }

    public boolean isPaused() {
        return isPaused;
    }

    public void setPaused(boolean paused) {
        isPaused = paused;
    }

    public void setTargetRadius(int targetRadius) {
        this.targetRadius = targetRadius;
    }
}

File: src/main/java/com/miilhozinho/chunkgenerator/data/GenState.kt
package com.miilhozinho.chunkgenerator.data

import com.hypixel.hytale.codec.Codec
import com.hypixel.hytale.codec.ExtraInfo
import com.hypixel.hytale.codec.KeyedCodec
import com.hypixel.hytale.codec.builder.BuilderCodec

class GenState {
    var worldName: String = ""
        private set
    var centerX: Int = 0
        private set
    var centerZ: Int = 0
        private set
    private var targetRadius = 0
    private var currentIndex = 0
    private var isPaused = true

    constructor()

    constructor(worldName: String, centerX: Int, centerZ: Int, targetRadius: Int) {
        this.worldName = worldName
        this.centerX = centerX
        this.centerZ = centerZ
        this.targetRadius = targetRadius
        this.currentIndex = 0
        this.isPaused = false
    }

    fun getTargetRadius(): Int {
        return targetRadius
    }

    fun getCurrentIndex(): Int {
        return currentIndex
    }

    fun setCurrentIndex(currentIndex: Int) {
        this.currentIndex = currentIndex
    }

    fun isPaused(): Boolean {
        return isPaused
    }

    fun setPaused(paused: Boolean) {
        isPaused = paused
    }

    fun setTargetRadius(targetRadius: Int) {
        this.targetRadius = targetRadius
    }

    companion object {
        val CODEC: BuilderCodec<GenState?> =
            BuilderCodec.builder(GenState::class.java) { GenState() }
                .append(
                    KeyedCodec("WorldName", Codec.STRING),
                    { state: GenState?, value: String, _: ExtraInfo? ->
                        state!!.worldName = value
                    },
                    { state: GenState?, _: ExtraInfo? -> state!!.worldName }).add()
                .append(
                    KeyedCodec("CenterX", Codec.INTEGER),
                    { state: GenState?, value: Int?, _: ExtraInfo? -> state!!.centerX = value!! },
                    { state: GenState?, _: ExtraInfo? -> state!!.centerX }).add()
                .append(
                    KeyedCodec("CenterZ", Codec.INTEGER),
                    { state: GenState?, value: Int?, _: ExtraInfo? -> state!!.centerZ = value!! },
                    { state: GenState?, _: ExtraInfo? -> state!!.centerZ }).add()
                .append(
                    KeyedCodec("TargetRadius", Codec.INTEGER),
                    { state: GenState?, value: Int?, _: ExtraInfo? ->
                        state!!.targetRadius = value!!
                    },
                    { state: GenState?, _: ExtraInfo? -> state!!.targetRadius }).add()
                .append(
                    KeyedCodec("CurrentIndex", Codec.INTEGER),
                    { state: GenState?, value: Int?, _: ExtraInfo? ->
                        state!!.currentIndex = value!!
                    },
                    { state: GenState?, _: ExtraInfo? -> state!!.currentIndex }).add()
                .append(
                    KeyedCodec("IsPaused", Codec.BOOLEAN),
                    { state: GenState?, value: Boolean?, _: ExtraInfo? ->
                        state!!.isPaused = value!!
                    },
                    { state: GenState?, _: ExtraInfo? -> state!!.isPaused }).add()
                .build()
    }
}